.github/workflows/carbonix_build.yml

```
@@ -446,3 +467,84 @@ jobs:
           $env:PATH_TO_S3 = '${{ needs.setup-s3-path.outputs.s3_path }}'
           echo "Uploading to: $env:PATH_TO_S3"
           aws s3 cp temp/ $env:PATH_TO_S3 --recursive
+
+  build-upload-zip:
+    runs-on: ubuntu-22.04
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'SKIP_BUILD') }}
+    needs: [process-ac, build-sitl, setup-s3-path]
+    steps:
+      - name: Extract folder name from s3_path
+        id: extract-folder-name
+        run: |
+          s3_path="${{ needs.setup-s3-path.outputs.s3_path }}"
+          folder_name=$(basename "$s3_path")
+          echo "folder_name=$folder_name" >> $GITHUB_ENV
+
+      - name: Create folder from s3_path
+        run: mkdir -p aircraft_zip
+
+      - name: List available artifacts
+        id: list-artifacts
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const artifacts = await github.rest.actions.listArtifactsForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+            return artifacts.data.artifacts.map(artifact => artifact.name);
+
+      - name: Filter artifacts
+        id: filter-artifacts
+        run: |
+          echo "Filtered artifacts:" > filtered_artifacts.txt
+          for artifact in ${{ steps.list-artifacts.outputs.result }}; do
+            if [[ $artifact =~ .*_AC_.*[0-9]+(\.[0-9]+)? ]]; then
+              echo $artifact >> filtered_artifacts.txt
+            fi
+          done
+          echo "sitl" >> filtered_artifacts.txt
+          cat filtered_artifacts.txt
+
+      - name: Download filtered artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ steps.filter-artifacts.outputs.result }}
+          path: aircraft_zip/
+      
+      - name: Delete unnecessary folders
+        run: |
+          rm -rf aircraft_zip/build-periph-*
+
+      - name: List downloaded files
+        run: ls -R aircraft_zip/
+      
+      - name: Install p7zip
+        run: sudo apt-get install -y p7zip-full
+
+      - name: 7zip the folder
+        run: |
+          cd aircraft_zip
+          7z a ../${{ env.folder_name }}.7z .
```

Spent some time tooling around to see what kind of compression performance I could get out of the standard tools. 7z seems to be a good balance of size and time. xz can get us down to 16k, but takes almost 3x longer. At compression level 8 (instead of 9), I get comparable size to 7z, but takes 60% longer.

Here's the tar invocation if you want it, but I think you picked well with 7z.

```suggestion
      - name: Archive the folder
        run: |
          cd aircraft_zip
          tar -cv --use-compress-program="xz -9ev --threads=0" -f ../${{ env.folder_name }}.7z .
```

---

Good to know the comparison. But I think 7z is small enough size and the time is better. So will stick with that.